src/opts/mod.rs

```diff
@@ -1210,8 +1211,8 @@ impl PoolConstraints {
     /// assert_eq!(opts.pool_opts().constraints(), PoolConstraints::new(0, 151).unwrap());
     /// # Ok(()) }
     /// ```
-    pub fn new(min: usize, max: usize) -> Option<PoolConstraints> {
-        if min <= max {
+    pub const fn new(min: usize, max: usize) -> Option<PoolConstraints> {
+        if min <= max && 0 < max {
```

Comment:
FWIW I'd usually expect this to be written the other way around

```suggestion
        if min <= max && max > 0 {
```